docs/blog/20230501-Daeraxa-MayUpdate.md

```diff
@@ -0,0 +1,161 @@
+
+<img src="./assets/flight-manual.png"/>
+
+Our last change is that relative links are now preserved against the repository of the package. For example `./LICENSE.md` => `https://github.com/pulsar-edit/package-frontend/LICENSE.md`.
+
+These were some much needed changes which should hopefully improve the overall experience of using the website.
+
+## Decoupling http handling for the package backend
+
+Lastly we have yet another [@confused-techie] update in the works, this time [improvements to the package backend](https://github.com/pulsar-edit/package-backend/pull/171) to make developing the backend more accessible to new contributors as well making tasks such as testing easier.
+
+Currently the handling for HTTP and logic is all mixed together in the backend which means that early every function within the backend has a signature like:
+
+```js
+aync function getPackages(res, req) {...}
+```
+
+Where `res` is the response object as provided by ExpressJS and `req` is the request object provided by ExpressJS.
+
+This also means when a function from the backend needs to return, it'll have a signature like:
+
+```js
+// On Success
+res.status(200).json(data);
+logger.httpLog(req, res);
+
+// On Failure
+await common.handleError(req, res, err);
+return;
+```
+
+Which has a few downsides; when testing a function to the backend a fake Response and Request object must be made that also matches the ones provided by ExpressJS and those objects have to be monitored for the functions behavior, or otherwise the API has to be tested as just that, an API by calling the ExpressJS module itself then checking the api returns for behavior.
+
+This also means that any new developers have to be familiar with ExpressJS to contribute and each function has to know the exact HTTP Status codes that have to be returned and in some circumstances concern themselves with header fields to set etc.
+
+The planned changes are to allow each function to return just a serverStatusObject (which is used frequently within the backend) so we change that same signature from above to:
+
+```js
+// On Success
+return {
+	ok: true,
+	content: data,
+};
+
+// On Failure
+return {
+	ok: false,
+	content: err,
+};
+```
+
+This has the upside of letting only the code that retrieves API requests deal with responding to them. The rest of the backend can work with just normal functions, without a care in the world about HTTP handling.
+The hope is that by lowering the amount of knowledge needed to contribute more people might be encouraged to get involved and comes with the added bonus that now to test we can just call functions.
```

Comment:
```suggestion
The hope is that by lowering the amount of knowledge needed to contribute, more people might be encouraged to get involved and comes with the added bonus that now to test we can just call functions.
```

This feels right in my head, but not sure if it's grammatically right lol

Reply:
Not quite, no.

“This change lowers the amount of knowledge needed to contribute, so we hope that more people might be encouraged to get involved. As a bonus, it makes the code much easier to test.“

Reply:
Sounds good, I've added this suggestion.